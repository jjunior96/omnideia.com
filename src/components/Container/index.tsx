import * as S from './styles';

type ContainerProps = {
  children: React.ReactNode;
};

const Container = ({ children }: ContainerProps) => {
  return <S.Container>{children}</S.Container>;
};

export default Container;
