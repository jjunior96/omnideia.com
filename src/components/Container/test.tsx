import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Container from '.';

describe('<Container />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <Container>
        <p>Test container</p>
      </Container>
    );

    expect(screen.getByText(/test container/i)).toBeInTheDocument();

    expect(container.firstChild).toHaveStyle({
      'max-width': '130rem'
    });

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c0 {
        width: 100%;
        max-width: 130rem;
        margin-left: auto;
        margin-right: auto;
      }

      <div
        class="c0"
      >
        <p>
          Test container
        </p>
      </div>
    `);
  });
});
