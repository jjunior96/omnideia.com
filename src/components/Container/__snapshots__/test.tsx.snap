// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding: 0 1.6rem;
}

<div
  class="c0"
>
  <p>
    Test container
  </p>
</div>
`;
